### DISCUSSION TITLE
egs_phd multiple spectrum scoring

### DISCUSSION THREAD

**Discussion by onrdm** [nrc-cnrc/EGSnrc#1050] [Questions] [answered]
I am using egs_phd to score the pulse height distribution in a region. What I would like to do is to extend this to a set of regions, each having its own spectrum file. I tried to manipulate the input file, but the app generates only one spectrum file. Any guidance on potential approaches to achieve the desired outcome of obtaining separate spectrum files for each individual region?

**Comment by rtownson** [comment, marked as answer]
If it's a short simulation, just write a little python script that runs the simulation many times, changing the region of interest. 

If this would take a long time because it's a long simulation, you'll have to edit the egs_phd code to add more spectra. Luckily, it's a nice short code that's pretty easy to read. Specifically, look through uses of the `spectrum` object, the whole `simulateSingleShower` function, and `outputResponse`. You'll need to duplicate the `spectrum` object for however many regions you would like. You could hard-code the regions into the code, or add a new input option that assigns each region in the list to a different spectrum file (then create spectra in a vector so that you can have any number). 

If you feel like you've done a good job, feel free to submit the modifications as a pull request! Make sure you are using `git`, and you'll have to copy your modifications to `HEN_HOUSE/user_codes/egs_phd/egs_phd.cpp`, since the `EGS_HOME` version of the code isn't version controlled.

**Reply by onrdm** [reply]
Thank you for the answer!

It will be a long simulation so I will try to edit the code following your guidance.

**Reply by rtownson** [reply]
If you have more questions while you work on this, feel free to re-open this discussion so that it's all in one place.

**Reply by onrdm** [reply]
what is the next step after modifying egs_phd.cpp and egs_phd.h files? I tried to compile it but it did not work. Are there any other files that need to be modified?

**Reply by rtownson** [reply]
While you're working on the changes, make sure you're in `egs_home/egs_phd`, not the `HEN_HOUSE`, because it will only compile properly in egs_home. What is the problem, does it compile? What is the error message?

**Reply by onrdm** [reply]
I am working on the egs_home/egs_phd. I don't know how but it finally compiled. Unfortunately, the code is not working as expected. I guess I need to work on that a little bit more...
